services/filemanager/filemanager.go

```diff
@@ -125,3 +138,10 @@ func GetProviderConfigFromEnv() map[string]interface{} {
 	}
 	return providerConfig
 }
+
+func getSafeTimeout(timeout *time.Duration) time.Duration {
```

Comment:
I am personally in favor of sane defaults, rather than ensuring/requiring that everything is specified and configured explicitly. 

This can be done on multiple levels: program & component.

In this case, I believe is a good idea to be able to create a manager like this:
```go
&S3Manager{
	Config:  GetS3Config(settings.Config),
}
```
and assume default timeout.

This is the same approach `http.Client` takes btw.

Maybe to use of the word `safe` is a bit misleading since it is one aspect of this function.
Note that, reading from `nil` pointer will cause a panic, so it is a good practice to explicitly check for that.